base/reshapedarray.jl

```diff
@@ -107,8 +107,19 @@ julia> reshape(1:6, 2, 3)
 """
 reshape
 
-reshape(parent::AbstractArray, dims::IntOrInd...) = reshape(parent, dims)
+reshape(parent::AbstractArray, dims::Union{Integer, AbstractUnitRange{<:Integer}, Colon}...) = reshape(parent, dims)
 reshape(parent::AbstractArray, shp::Tuple{Union{Integer,OneTo}, Vararg{Union{Integer,OneTo}}}) = reshape(parent, to_shape(shp))
+reshape(parent::AbstractArray, shp::Tuple) = reshape(parent, map(_toshape, shp)::Tuple{Vararg{Union{Int,Colon}}})
```

Comment:
This uses `Tuple` instead of a union of valid types to allow OffsetArrays to define methods for AbstractUnitRange arguments that return `OffsetArray`s